cds_migrator_kit/videos/weblecture_migration/users/load.py

```diff
@@ -0,0 +1,142 @@
+logger_submitters = logging.getLogger("submitters")
+
+
+class VideosSubmitterLoad(Load):
+    """Submitter load class."""
+
+    def __init__(
+        self,
+        missing_users_dir=None,
+        dry_run=False,
+    ):
+        """Constructor."""
+        self.dry_run = dry_run
+        self.missing_users_dir = missing_users_dir
+        self.dry_run = dry_run
+
+    def _load(self, entry):
+        """Load users."""
+        self._owner(entry)
+
+    def _validate(self, entry):
+        """Validate data before loading."""
+        if not entry:
+            return False
+        return True
+
+    def _owner(self, json_entry):
+        """Fetch or create owner."""
+        email = json_entry.get("submitter")
+        if not email:
+            return
+        try:
+            user = User.query.filter_by(email=email).one()
+            logger_submitters.info(f"User {email} exists.")
+
+            return user.id
+        except NoResultFound:
+            if not self.dry_run:
+                user_id = self._create_owner(email)
+                return user_id
+
+    def _create_owner(self, email_addr):
+        """Create owner from legacy data.
+
+        Every record needs an owner assigned in parent.access.owned_by
+        therefore we need to create dummy accounts
+        """
+
+        def get_person_old_db(email):
+            missing_users_dump = os.path.join(
+                self.missing_users_dir, "missing_users.json"
+            )
+            with open(missing_users_dump) as json_file:
+                missing = json.load(json_file)
+            person = next((item for item in missing if item["email"] == email), None)
+
+            return person
+
+        user_api = MigrationUserAPI()
+        person_old_db = get_person_old_db(email_addr)
+
+        person_id = None
+        displayname = None
+        username = None
+        existing_identity = None
+        extra_data = {"migration": {}}
+
+        # check if email exists in legacy db
+        # to obtain more info needed to create an account (full name, username)
+
+        # if user email cannot be found anywhere, create a dummy account without
+        # any UserIdentity attached - since we cannot provide identity info -
+        # the consequence is that user will not be able to login to this account
+        # if they come back to CERN
+
+        if person_old_db:
+            names = "".join(person_old_db["displayname"].split())
+            username = names.lower().replace(".", "")
+            if not username:
+                username = f'MIGRATED{email_addr.split("@")[0].replace(".", "")}'
+                username = re.sub(r"\W+", "", username)
+            displayname = person_old_db["displayname"]
+            extra_data["migration"]["source"] = "LEGACY DB, PERSON ID MISSING"
+            logger_submitters.warning(f"User {email_addr} found in legacy DB")
+        else:
+            username = email_addr.split("@")[0].replace(".", "")
```

Comment:
this is fine here. I think slowly we can create migration issue where  we document all these cases that don't fail but we log a warning so we can check these after the migration.